### PR TITLE
add cart feature

### PR DESCRIPTION
**Pull request by Anhphung14**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -15,6 +16,7 @@ app.use(express.json({ limit: "50mb" }));
 
 //ROUTER
 app.use("/api/v1/categories", categoryRouter);
+app.use("/api/v1/cart", cartRouter);
```
Comment (phucnguyn-203) [inline server/app.js:19]: /api/v1/carts

File: server/controllers/cartControllers.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const Cart = require("../models/cartModel");
+const Category = require("../models/categoryModel");
```
Comment (phucnguyn-203) [inline server/controllers/cartControllers.js:2]: xoá dòng này

File: server/controllers/cartControllers.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const catchAsync = require("../utils/catchAsync");
+
+exports.getAllCart = catchAsync(async (req, res) => {
+    const cart = await Cart.find();
```
Comment (phucnguyn-203) [inline server/controllers/cartControllers.js:6]: const carts

File: server/controllers/cartControllers.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    res.status(200).json({
+        status: "success",
+        data: cart,
```
Comment (phucnguyn-203) [inline server/controllers/cartControllers.js:9]: data: carts

File: server/controllers/cartControllers.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+exports.updateCart = catchAsync(async (req, res) => {
+    const cart = await Cart.findByIdAndUpdate(req.params.id, req.body, { new: true, runValidators: true });
+    res.status(200).json({
+        status: "success",
+        data: cart,
+    });
+});
+
+exports.deleCart = catchAsync(async (req, res) => {
```
Comment (phucnguyn-203) [inline server/controllers/cartControllers.js:37]: deleteCart

File: server/routes/cartRouter.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const express = require("express");
+const router = express.Router();
+
+const { getAllCart, getOneCart, createCart, deleCart, updateCart } = require("../controllers/cartControllers");
```
Comment (phucnguyn-203) [inline server/routes/cartRouter.js:4]: deleteCart

File: server/routes/cartRouter.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+router.get("/", getAllCart);
+router.get("/:id", getOneCart);
+router.post("/", createCart);
+router.delete("/:id", deleCart);
```
Comment (phucnguyn-203) [inline server/routes/cartRouter.js:9]: deleteCart

File: server/models/cartModel.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
+
+const cartSchema = new Schema({
+    items: [
+        {
+            productId: {
```
Comment (phucnguyn-203) [inline server/models/cartModel.js:7]: `product: {
      type: Schema.Types.ObjectId
      ref: "Product"
      required: true
}`

File: server/models/cartModel.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            },
+            total: {
+                type: Number,
+                require: true,
+            },
+        },
+    ],
+    customer_id: {
+        type: String,
+        require: true,
```
Comment (phucnguyn-203) [inline server/models/cartModel.js:24]: unique: true

File: server/models/cartModel.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+                require: true,
+            },
+            total: {
+                type: Number,
+                require: true,
+            },
+        },
+    ],
+    customer_id: {
+        type: String,
```
Comment (phucnguyn-203) [inline server/models/cartModel.js:23]: type: Schema.Types.ObjectId
ref: "User"

File: server/app.js (hunk 1 above)
Comment (Anhphung14) [reply server/app.js:19]: done

File: server/controllers/cartControllers.js (hunk 2 above)
Comment (Anhphung14) [reply server/controllers/cartControllers.js:2]: done

**Pull request merged**